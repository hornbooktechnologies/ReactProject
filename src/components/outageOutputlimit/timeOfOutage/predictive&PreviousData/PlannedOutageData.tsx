import React from 'react'
import { PlannedOutageDataModel } from '../../../../models/outageOutputLimit/PlannedOutageDataModel'
import { useTranslation } from 'react-i18next'
import { getDiffOfValues, IsappearTooltip, prefixSign } from '../../../../utils/utils'
import Tooltip from '../../../common/Tooltip'

const PlannedOutageData = ({
  planTitle,
  planVal,
  planpreviousVal,
  suffix,
  subject,
  className,
  iconClassName,
  variation
}: PlannedOutageDataModel) => {
  const { i18n } = useTranslation()
  let diffPlan = getDiffOfValues(planVal,planpreviousVal)

  return (
    <>
      <div className={`uk-flex uk-flex-middle ${className}`}>
        <span className="uk-up-icon uk-text-center uk-inline mr-5">
          <i className={`jera icn-24 ${iconClassName} uk-position-center`}></i>
        </span>
        <div>
          <p className="uk-margin-remove font-14">{planTitle}</p>
          <span className="mb-0 uk-text-bold font-14 mt-5">
            <span
              className={`${IsappearTooltip(diffPlan)}`}>
              {prefixSign(planVal,variation)}
              {planVal}
              <span className="font-8 uk-text-normal ml-3">{suffix}</span>
              {diffPlan === true ? (
                <Tooltip
                  prefix={prefixSign(planpreviousVal,variation)}
                  previousDayValue={planpreviousVal}
                  unit={suffix}
                />
              ) : null}
            </span>
            <span className="font-8 uk-text-normal ml-3">
              <span className="uk-text-bold font-12">
              ({subject}
                <span className="font-8">
                  {i18n.t('OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT')}
                </span>
                )
              </span>
            </span>
          </span>
        </div>
      </div>
    </>
  )
}

export default PlannedOutageData
