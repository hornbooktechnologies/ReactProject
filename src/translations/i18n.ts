import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
