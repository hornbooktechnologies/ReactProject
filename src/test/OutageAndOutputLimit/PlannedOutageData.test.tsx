import React from "react";
import renderer from "react-test-renderer";
import PlannedOutageData from "../../components/outageOutputlimit/timeOfOutage/predictive&PreviousData/PlannedOutageData";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe('Component: PlannedOutageData', () => {
  it('PlannedOutageData renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <PlannedOutageData />
        </I18nextProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
