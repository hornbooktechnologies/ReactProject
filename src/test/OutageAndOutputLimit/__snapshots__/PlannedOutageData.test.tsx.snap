// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PlannedOutageData PlannedOutageData renders correctly 1`] = `
<div
  className="uk-flex uk-flex-middle undefined"
>
  <span
    className="uk-up-icon uk-text-center uk-inline mr-5"
  >
    <i
      className="jera icn-24 undefined uk-position-center"
    />
  </span>
  <div>
    <p
      className="uk-margin-remove font-14"
    />
    <span
      className="mb-0 uk-text-bold font-14 mt-5"
    >
      <span
        className=""
      >
        <span
          className="font-8 uk-text-normal ml-3"
        />
      </span>
      <span
        className="font-8 uk-text-normal ml-3"
      >
        <span
          className="uk-text-bold font-12"
        >
          (
          <span
            className="font-8"
          >
            OUTAGE_AND_OUTPUT_LIMIT.OUTAGE_PRICE_SUB_UNIT
          </span>
          )
        </span>
      </span>
    </span>
  </div>
</div>
`;
